oauthproxy_test.go

```diff
@@ -1686,3 +1686,43 @@ func Test_noCacheHeadersDoesNotExistsInResponseHeadersFromUpstream(t *testing.T)
 		assert.Equal(t, "", rec.Header().Get(k))
 	}
 }
+
+func TestIPCIDRSet(t *testing.T) {
```

Comment:
This may be better converted to a table style test, that way we could add some tests for the error cases as well, currently this only covers the happy paths. Eg we could see what happens if the `RealClientIP` cannot be found from the header